tools/Tool.cc

```diff
@@ -51,6 +57,61 @@ static bool outputIsColor() {
     return sColor;
 }
 
+static bool tokenize(const char* input, deque<string> &args)
+{
+    string str(input);
+    int start = 0;
+    for( size_t i=0; i<str.length(); i++){
+        char c = str[i];
+        if( c == ' ' ) {
+            if(i - start > 1) {
+                args.push_back(str.substr(start, i - start));
+            }
+            
+            while(str[i] == ' ') {
+                i++;
+            }
+            
+            start = i;
+            i--;
+        } else if(c == '\"' ) {
+            i++;
+            while( i < str.size() - 1 && str[i] != '\"' ){ i++; }
+            if(str[i] != '\"') {
+                return false;
+            }
+            
+            args.push_back(str.substr(start + 1, i - start - 1));
+            start = i;
+        } else if(c == '\r' || c == '\n') {
+            break;
+        } else if(!isascii(c)) {
```

Comment:
This seems like the wrong way to treat non-ASCII bytes. For example, `foo¿bar` will produce the argument `bar` because `start` moves forward past the `foo`. (No wait, actually it'll produce invalid UTF-8 because the subsequent byte(s) of the UTF-8 sequence for `¿` will be emitted but not the first.